deps/wrapper.cc

```diff
@@ -57,25 +49,50 @@ std::string get(ObjectID object_id) {
 
     std::shared_ptr<RayObject> result = results[0];
     if (result == nullptr) {
-        return "\0";
+        return nullptr;
     }
 
-    std::string data = (std::string) reinterpret_cast<char *>(result->GetData()->Data());
-    return data;
+    return result->GetData();
 }
 
 std::string ToString(ray::FunctionDescriptor function_descriptor)
 {
     return function_descriptor->ToString();
 }
 
+namespace jlcxx
+{
+    // Needed for upcasting
+    template<> struct SuperType<LocalMemoryBuffer> { typedef Buffer type; };
+
+    // Disable generated constructors
+    // https://github.com/JuliaInterop/CxxWrap.jl/issues/141#issuecomment-491373720
+    template<> struct DefaultConstructible<LocalMemoryBuffer> : std::false_type {};
+
+    // Custom finalizer to show what is being deleted. Can be useful in tracking down
+    // segmentation faults due to double deallocations
+    // https://github.com/JuliaInterop/CxxWrap.jl/tree/main#overriding-finalization-behavior
+    template<typename T>
+    struct Finalizer<T, SpecializedFinalizer>
+    {
+        static void finalize(T* to_delete)
+        {
+            std::cout << "calling delete on: " << to_delete << std::endl;
+            delete to_delete;
+        }
+    };
+}
```

Comment:
is htis still necessary?  IIUC it was used to debug the segfaults but those are taken care of now...